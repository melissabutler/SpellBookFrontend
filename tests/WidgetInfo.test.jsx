import React from "react";
import { describe, it, expect, vi } from "vitest";
import { render, screen} from '@testing-library/react'
import InfoWidget from "../src/Components/WidgetInfo";

describe("InfoWidget", function() {

    it('renders the InfoWidget component', async () => {
        vi.mock('react', async (importOriginal) => {
            const react = await importOriginal('react');
            return {
                ...react,
                useContext: () => ({ currentUser: null}),
                useRef: () => { target: null }
        }})

        vi.mock('react-router-dom', async () => {
            const mod = await vi.importActual('react-router-dom');
            return {
                ...mod, 
                useNavigate: () => "/",
                BrowserRouter: () => "",
                Link: () => ""
            }
        })
    
        render(
            <InfoWidget />)
        screen.debug();

    
      
    });

        it('matches snapshot', () => {
            const { asFragment } = render(<InfoWidget />);
            expect(asFragment()).toMatchSnapshot();
        });



})
   