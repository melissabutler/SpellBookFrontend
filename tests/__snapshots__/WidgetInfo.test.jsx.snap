// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InfoWidget > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Widget undefined container"
  />
</DocumentFragment>
`;
